src/api/shopper-insights/component.test.js

```diff
@@ -67,20 +65,87 @@ describe("shopper insights component - getRecommendedPaymentMethods()", () => {
         },
       });
 
-    expect(callRestAPI).toHaveBeenCalled();
+    expect(request).toHaveBeenCalled();
     expect(recommendedPaymentMethods).toEqual({
       isPayPalRecommended: true,
       isVenmoRecommended: false,
     });
     expect.assertions(2);
   });
 
+  test("should get recommended payment methods from memoized request for the exact same payload", async () => {
+    const shopperInsightsComponent = getShopperInsightsComponent();
+    const payload = {
+      customer: {
+        email: "email-1.0@test.com",
+        phone: {
+          countryCode: "1",
+          nationalNumber: "2345678901",
+        },
+      },
+    };
+    const response1 =
+      await shopperInsightsComponent.getRecommendedPaymentMethods(payload);
+    expect(request).toHaveBeenCalled();
+    expect(request).toHaveBeenCalledTimes(1);
+    const response2 =
+      await shopperInsightsComponent.getRecommendedPaymentMethods(payload);
+
+    expect(request).toHaveBeenCalled();
+    // This should not change as the payload is same
+    expect(request).toHaveBeenCalledTimes(1);
+    expect(response1).toEqual({
+      isPayPalRecommended: true,
+      isVenmoRecommended: false,
+    });
+    expect(response2).toEqual({
+      isPayPalRecommended: true,
+      isVenmoRecommended: false,
+    });
+    expect.assertions(6);
+  });
```

Comment:
awesome test coverage! The way you structured this test made it super easy to understand why the change is being made 🏅